### PR TITLE
AIP-72: Adding client tests for all task instance operations

### PR DESCRIPTION
**Pull request by amoghrajesh**
We have basic sanity tests for every type of operation in the SDK API client, like variables, xcom. We do not have this for task instances, adding it through this PR. The aim is to maintain client sanity while developing new APIs for the execution API

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: task_sdk/tests/api/test_client.py [hunk 1]
```diff
@@ -82,6 +85,88 @@ def make_client(transport: httpx.MockTransport) -> Client:
...
+
+    def test_task_instance_start(self):
+        # Simulate a successful response from the server that starts a task
+        ti_id = uuid6.uuid7()
+
+        def handle_request(request: httpx.Request) -> httpx.Response:
+            if request.url.path == f"/task-instances/{ti_id}/state":
+                return httpx.Response(
+                    status_code=204,
+                )
```
Comment (ashb) [inline task_sdk/tests/api/test_client.py]: We should probably assert something about the request body/payload here I think.

File: task_sdk/tests/api/test_client.py [hunk 2]
```diff
@@ -82,6 +85,88 @@ def make_client(transport: httpx.MockTransport) -> Client:
...
+        client = make_client(transport=httpx.MockTransport(handle_request))
+        try:
+            msg = DeferTask(
+                classpath="airflow.providers.standard.triggers.temporal.DateTimeTrigger",
+                trigger_kwargs={"moment": "2024-11-07T12:34:59Z", "end_from_trigger": False},
+                next_method="execute_complete",
+            )
+            client.task_instances.defer(ti_id, msg)
+        except Exception as e:
+            pytest.fail(f"Unexpected error occurred: {e}")
```
Comment (ashb) [inline task_sdk/tests/api/test_client.py]: This isn't needed, pytest will already mark it as failed if the test fails

**Review by ashb — APPROVED** [review]
Nice. Lets also check something about the request bodies where we send one, then this is good to go too

File: task_sdk/tests/api/test_client.py (hunk 1 above)
Comment (amoghrajesh) [reply task_sdk/tests/api/test_client.py]: Yeah makes sense. Adding some validation

Comment (amoghrajesh) [reply task_sdk/tests/api/test_client.py:103]: Yeah makes sense. Adding some validation

File: task_sdk/tests/api/test_client.py (hunk 2 above)
Comment (amoghrajesh) [reply task_sdk/tests/api/test_client.py]: Yeah, removed all the except blocks

File: task_sdk/tests/api/test_client.py [hunk 3]
```diff
@@ -192,112 +273,3 @@ def handle_request(request: httpx.Request) -> httpx.Response:
 
         result = client.variables.set(key="test_key", value="test_value", description="test_description")
         assert result == {"ok": True}
-
-
-class TestXCOMOperations:
```
Comment (kaxil) [inline task_sdk/tests/api/test_client.py:197]: Looks like this removal wasn't intended

**Pull request merged**